aces/idt/generators/base_generator.py

```diff
@@ -0,0 +1,745 @@
+        process.
+
+        Returns
+        -------
+        :class:`NDArray` or None
+            Samples produced by the colour checker sampling process.
+        """
+
+        return self._samples_analysis
+
+    def sample(self):
+        """
+        Sample the images from the *IDT* specification.
+        """
+
+        logger.info('Sampling "IDT" specification images...')
+
+        settings = Structure(**common.SETTINGS_SEGMENTATION_COLORCHECKER_CLASSIC)
+        working_width = settings.working_width
+        working_height = settings.working_height
+
+        def _reformat_image(image):
+            """Reformat given image."""
+
+            return reformat_image(
+                image, settings.working_width, settings.interpolation_method
+            )
+
+        rectangle = as_int32_array(
+            [
+                [working_width, 0],
+                [working_width, working_height],
+                [0, working_height],
+                [0, 0],
+            ]
+        )
+
+        self._samples_analysis = deepcopy(self.project_settings.data)
+        # Baseline exposure value, it can be different from zero.
+        if 0 not in self.project_settings.data[DataFolderStructure.COLOUR_CHECKER]:
+            EVs = sorted(
+                self.project_settings.data[DataFolderStructure.COLOUR_CHECKER].keys()
+            )
+            self._baseline_exposure = EVs[len(EVs) // 2]
+            logger.warning(
+                "Baseline exposure is different from zero: %s", self._baseline_exposure
+            )
+
+        paths = self.project_settings.data[DataFolderStructure.COLOUR_CHECKER][
+            self._baseline_exposure
+        ]
+
+        with working_directory(self.project_settings.working_directory):
+            logger.info(
+                'Reading EV "%s" baseline exposure "ColourChecker" from "%s"...',
+                self._baseline_exposure,
+                paths[0],
+            )
+            image = _reformat_image(read_image(str(paths[0])))
+
+        (
+            rectangles,
+            clusters,
+            swatches,
+            segmented_image,
+        ) = segmenter_default(
+            image,
+            additional_data=True,
+            **common.SETTINGS_SEGMENTATION_COLORCHECKER_CLASSIC,
+        ).values
+
+        quadrilateral = rectangles[0]
+
+        self._image_colour_checker_segmentation = np.copy(image)
+        cv2.drawContours(
+            self._image_colour_checker_segmentation, swatches, -1, (1, 0, 1), 3
+        )
+        cv2.drawContours(
+            self._image_colour_checker_segmentation, clusters, -1, (0, 1, 1), 3
+        )
+
+        data_detection_colour_checker_EV0 = sample_colour_checker(
+            image, quadrilateral, rectangle, common.SAMPLES_COUNT_DEFAULT, **settings
+        )
+
+        # Disabling orientation as we now have an oriented quadrilateral
+        settings.reference_values = None
+
+        # TODO Again are we using this?
+        # Flatfield
```

Comment:
For v2 yeah! :)

Reply:
Ok so lets include flatfield through out while we are writing the cleanup, I didnt know if this was old, new, or proposed lol :)

Reply:
Ok removed TODOs and matched as a constant like the other folders